main.go

```diff
@@ -95,6 +96,8 @@ func main() {
 	var outputMode string
 	var serverNameOverride string
 
+	ioController := io.NewIOController()
```

Comment:
io controller sounds like it's more than logging, is that the purpuse of this new pkg?

Reply:
Yes, right now we have a logger package that controls every in/out operation and the logger. With this approach we will move into a `ioCtrl` that supports all previous behavior

Reply:
I dont know why we cant use `log.Logger()` instead io. At the end only a logger will control al this logic

Reply:
A logger refers to the debug/info logs that the developer uses for troubleshooting. IO refers to more than that. It refers to all the output that can go to STDOUT STDIN STDERR and how it handles them. 

If we continue with the logger. We will still have the problem to know if a log is written to stdout, stderr a file or where. This refactor comes to remove all that kind of friction